### PR TITLE
Add `is_derived` to `DirectiveFinder`

### LINKED ISSUE #2464: Create a dedicated method to check if a field is derived.
We must create a `schema_field.is_derived()` method to `DirectiveFinder` to cut down on the number of places where we hardcode the string "derivedFrom" across the codebase to detect if a field is derived or not.

_Originally posted by @lutter in https://github.com/graphprotocol/graph-node/pull/2352#discussion_r631329917_

### PR DESCRIPTION
**Pull request by vrmiguel**
Closes #2464 

Thanks @Xunjin for the help with this PR 😊

### CODE REVIEW

**Comment by vrmiguel** [comment]
I think it'd be nice to implement some unit tests for things relating to `DirectiveFinder` but I'm having difficulties working with `ObjectType`, `Field` and `Directive`

Any help with that would be very appreciated :-)

**Comment by tilacog** [comment]
I believe that an interesting test approach would involve [parsing](https://docs.rs/graphql-parser/0.3.0/graphql_parser/index.html) a simple graphql schema that contains some directives and then call the new functions over that structure, asserting that the given directives were found and associated with the expected fields.

**Comment by tilacog** [comment]
LGTM

File: graph/src/data/graphql/ext.rs [hunk 1]
```diff
@@ -266,10 +273,78 @@ impl DirectiveFinder for Field {
...
+mod directive_finder_tests {
+    use graphql_parser::parse_schema;
+
+    use super::*;
+
+    const SCHEMA: &str = "
+    type BuyEvent implements Event @derivedFrom(field: \"buyEvent\") {
+        id: ID!,
+        transaction: Transaction! @derivedFrom(field: \"buyEvent\")
+    }";
```
Comment (tilacog) [inline graph/src/data/graphql/ext.rs:306]: Just minor note:
As far as I can tell, the `derivedFrom` directive is applied to fields, not to types. 
Thus, I'd suggest removing the directive from the type and focusing the test only on fields marked by it.

**Review by tilacog — APPROVED** [review]

**Pull request merged**